l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -141,11 +322,53 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, ReentrancyGuard {
         uint256 _l2TxGasLimit,
         uint256 _l2TxGasPerPubdataByte
     ) external payable returns (bytes32 l2TxHash) {
-        l2TxHash = deposit(_l2Receiver, _l1Token, _amount, _l2TxGasLimit, _l2TxGasPerPubdataByte, address(0));
+        l2TxHash = deposit(
+            ERA_CHAIN_ID,
+            _l2Receiver,
+            _l1Token,
+            msg.value,
+            _amount,
+            _l2TxGasLimit,
+            _l2TxGasPerPubdataByte,
+            address(0)
+        );
+    }
+
+    /// @notice Legacy deposit method with no chainId, use another `deposit` method instead.
+    /// @dev Initiates a deposit by locking funds on the contract and sending the request
+    /// of processing an L2 transaction where tokens would be minted
+    /// @param _l2Receiver The account address that should receive funds on L2
+    /// @param _l1Token The L1 token address which is deposited
+    /// @param _amount The total amount of tokens to be bridged
+    /// @param _l2TxGasLimit The L2 gas limit to be used in the corresponding L2 transaction
+    /// @param _l2TxGasPerPubdataByte The gasPerPubdataByteLimit to be used in the corresponding L2 transaction
+    /// @return l2TxHash The L2 transaction hash of deposit finalization
+    /// @param _refundRecipient The address on L2 that will receive the refund for the transaction.
+    /// NOTE: the function doesn't use `nonreentrant` and `senderCanCallFunction` modifiers,
+    /// because the inner method does.
+    function deposit(
+        address _l2Receiver,
+        address _l1Token,
+        uint256 _amount,
+        uint256 _l2TxGasLimit,
+        uint256 _l2TxGasPerPubdataByte,
+        address _refundRecipient
+    ) external payable returns (bytes32 l2TxHash) {
+        l2TxHash = deposit(
+            ERA_CHAIN_ID,
+            _l2Receiver,
+            _l1Token,
+            msg.value,
+            _amount,
+            _l2TxGasLimit,
+            _l2TxGasPerPubdataByte,
+            _refundRecipient
+        );
     }
 
     /// @notice Initiates a deposit by locking funds on the contract and sending the request
     /// of processing an L2 transaction where tokens would be minted
+    /// only used for eth based chains
```

Comment:
Not, we can only keep this for eth based chains. The deposit process got a bit more complicated due to the approve process. 

So if you are depositing to a non eth base chain, the flow is as follows: 

User->Bridgehub->BaseTokenBridge<-Bridgehub->SecondBridge(e.g. this bridge) <- Bridgehub ->Mailbox <-Bridgehub ->SecondBridge (to confirm tx). 

This is needed because both the BaseTokenBridge and the SecondBridge need to know the original msg.sender (as they request deposits from both). This can only be done if both trust the Bridgehub, and the bridgehub tells them the caller.